pilot/pkg/proxy/envoy/v2/discovery.go

```diff
@@ -194,6 +199,13 @@ func NewDiscoveryServer(
 		return nil
 	}
 
+	instanceUpdateHandler := func(instance *model.ServiceInstance, event model.Event) {
+		out.updateProxyServiceInstances(instance)
+	}
+	if err := ctl.AppendInstanceHandler(instanceUpdateHandler); err != nil {
```

Comment:
Why do we need another Instance handler  to trigger push as we already have one in line 197, which will trigger a full push.

Reply:
Well, up until now the instance handler wasn't even invoked (see 3fdd00e). It's possible that the fix in 3fdd00e is sufficient to resolve the issue (I'm not sure what's the outcome of `DiscoveryServer.clearCache()`), and the additional instance handler added in 4da238e is redundant. I'll test and report back.

Reply:
We cannot call clearCache if it wasn't already being called. clearCache triggers a full push to the entire cluster. That means any time a new instances comes up a full push goes to every proxy